osu.Game/Screens/Ranking/Expanded/StarRatingDisplay.cs

```diff
@@ -78,32 +89,65 @@ private void load(OsuColour colours, BeatmapDifficultyCache difficultyCache)
                             Icon = FontAwesome.Solid.Star,
                             Colour = Color4.Black
                         },
-                        new OsuTextFlowContainer(s => s.Font = OsuFont.Numeric.With(weight: FontWeight.Black))
+                        textFlow = new OsuTextFlowContainer(s => s.Font = OsuFont.Numeric.With(weight: FontWeight.Black))
                         {
                             Anchor = Anchor.CentreLeft,
                             Origin = Anchor.CentreLeft,
                             AutoSizeAxes = Axes.Both,
                             Direction = FillDirection.Horizontal,
                             TextAnchor = Anchor.BottomLeft,
-                        }.With(t =>
-                        {
-                            t.AddText($"{wholePart}", s =>
-                            {
-                                s.Colour = Color4.Black;
-                                s.Font = s.Font.With(size: 14);
-                                s.UseFullGlyphHeight = false;
-                            });
-
-                            t.AddText($"{separator}{fractionPart}", s =>
-                            {
-                                s.Colour = Color4.Black;
-                                s.Font = s.Font.With(size: 7);
-                                s.UseFullGlyphHeight = false;
-                            });
-                        })
+                        }
                     }
                 }
             };
         }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            Current.BindValueChanged(_ => updateDisplay(), true);
+            FinishTransforms(true);
+        }
+
+        private void updateDisplay()
+        {
+            const double duration = 400;
+            const Easing easing = Easing.OutQuint;
+
+            double stars = Current.Value?.Stars ?? 0.00f;
+
+            var starRatingParts = stars.ToString("0.00", CultureInfo.InvariantCulture).Split('.');
+            string wholePart = starRatingParts[0];
+            string fractionPart = starRatingParts[1];
+            string separator = CultureInfo.CurrentCulture.NumberFormat.NumberDecimalSeparator;
+
+            if (Current.Value == null)
+                background.FadeColour(Color4.SlateGray.Opacity(0.3f));
```

Comment:
In the case of metadata display (which this is intended for), wouldn't it make more sense to fade the rating out entirely than to show a faintly visible 0? That seems kind of confusing from a UX sense. It's data that's not ready to be displayed, so I don't think it should be. It almost looks like a bug.

Reply:
I was not sure about doing that at first, but now that I re-remember the existence of `AutoSizeDuration`, can make something out of it.

Reply:
I'd probably go for a `-` when data is unavailable. Showing nothing at all doesn't seem correct.

Reply:
A plain `-` sprite text placeholder on the metadata display itself? or inside the `StarRatingDisplay` in here?

Reply:
I'm not really sure, would have to see how things look. This might be a bit complicated based on how it's currently split up.

Reply:
There's no test covering this state.

Reply:
It's kind of already covered as it starts with null difficulty, but I'll add an explicit test step.

Reply:
> In the case of metadata display (which this is intended for), wouldn't it make more sense to fade the rating out entirely than to show a faintly visible 0? That seems kind of confusing from a UX sense. It's data that's not ready to be displayed, so I don't think it should be. It almost looks like a bug.

I've tried this approach + some fancy transforms and it looks to be good enough as a simple alternative way for the time being until the whole screen gets redesigned:

![CleanShot 2021-05-14 at 15 24 59](https://user-images.githubusercontent.com/22781491/118270445-c23a6d80-b4c8-11eb-93a0-ea3fb007f6a1.gif)